worldedit-core/src/main/java/com/sk89q/worldedit/world/block/BlockType.java

```diff
@@ -247,6 +247,21 @@ public boolean hasProperty(PropertyKey key) {
         return this.settings.propertiesMapArr.length > ordinal && this.settings.propertiesMapArr[ordinal] != null;
     }
 
+    /**
+     * {@return whether this block type has a property with given key of the given type}
+     *
+     * @param key          the key identifying the property
+     * @param propertyType the expected type of the property
+     * @since TODO
+     */
+    public boolean hasPropertyOfType(PropertyKey key, Class<?> propertyType) {
+        int ordinal = key.getId();
+        Property<?> property;
+        return this.settings.propertiesMapArr.length > ordinal
```

Comment:
Does this definitely make sense to have? If the key ID is dynamic to the original block state, that may not work, and if it's global that wouldn't work either?

Reply:
Our `PropertyKey` and its ID are unique, so they are at the same position in each `propertiesMapArr`. But the actual property class stored there can have different types in different arrays.

Reply:
Okay cool, so just a quick and dirty check so we don't need to check further